Libplanet.Net/Consensus/Context.cs

```diff
@@ -233,7 +233,14 @@ public VoteSet VoteSet(int round)
                 ? new VoteSet(Height, round, b.Hash, _validators)
                 : throw new NullReferenceException(
                     $"Cannot create a {nameof(Libplanet.Consensus.VoteSet)} for a null block");
-            _messageLog.GetPreCommits(round).ForEach(commit => voteSet.Add(commit.PreCommit));
+            _messageLog.GetPreCommits(round)
+                .ForEach(commit =>
+                {
+                    if (commit.PreCommit.BlockHash.Equals(block.Hash))
+                    {
+                        voteSet.Add(commit.PreCommit);
+                    }
+                });
```

Comment:
```csharp
            _messageLog.GetPreCommits(round)
                .Where(commit => commit.PreCommit.BlockHash.Equals(block.Hash))
                .ForEach(commit => voteSet.Add(commit.PreCommit));
```

I'd go further with `LINQ`. 😗

Reply:
I have avoided suggesting approach because `GetPreCommits().Where()` results have to casted in `List<T>`, `.ToList()`, again.